test/test_caching.py

```diff
@@ -0,0 +1,9 @@
+from sql_metadata import Parser
+
+
+def test_cleared_cache():
+    parser = Parser("Select * from test")
+    assert parser.tables == ["test"]
+
+    parser.query = "Select * from test2"
```

Comment:
Now let's check that such assignment raises an exception.

Reply:
Applied.